modules/Product.js

```diff
@@ -70,13 +70,17 @@ class Product {
                 for (const start_vertex of response[0].objects) {
                     // eslint-disable-next-line
                    const virtualGraph = this.zeroKnowledge(await this.graphStorage.findTraversalPath(start_vertex, depth));
+                    const lastDatasetIndex = start_vertex.datasets.length - 1;
+                    const datasetId = start_vertex.datasets[lastDatasetIndex];
                     responseData.push({
                         start: start_vertex._key,
+                        batch: start_vertex[datasetId].data,
                         data: virtualGraph.data,
                     });
                 }
                 resolve(responseData);
             }).catch((error) => {
+                console.log(error);
```

Comment:
Always welcome.

Reply:
@kipliklotrika I changed this to use logger, please let me know if that is correct now?